### PR TITLE
chore(deps): replace/bump github.com/maas/gomaasclient

### PR DESCRIPTION
**Pull request by troyanov**
Replace github.com/maas/gomaasclient with github.com/canonical/gomaasclient 
Bump version from 0.5.0 to 0.6.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,14 +1,16 @@
 module terraform-provider-maas
 
-go 1.20
```
Comment (skatsaounis) [inline go.mod:3]: Could you please revert this change? go version bump + toolchain

**Review by skatsaounis — CHANGES_REQUESTED** [review]

**Review by skatsaounis — APPROVED** [review]
LGTM +1

File: go.mod (hunk 1 above)
Comment (troyanov) [reply go.mod]: Hm, it seems that this change was here due to https://github.com/canonical/terraform-provider-maas/pull/179
> go: github.com/bflad/tfproviderlint@v0.30.0 requires go@1.21, but 1.20 is requested
